### PR TITLE
PR to configure frontend environment with linting

### PR DESCRIPTION
**Pull request by christianmckinnon**
## Description

- Added .eslintrc.js file to configure the linter using ESLint and Prettier.
- Added .env file with PORT=3001 to follow best practice.
- Added linter plugins and configs to "devDependencies" in package.json.
- Reset the Dockerfile to the original configuration such that it runs on PORT=3001.
- Docstrings have been added to several files for organizational purposes.

## Related Issue

[Jira Issue 13](https://oregonstate-team-a4gvoaif.atlassian.net/jira/software/projects/KAN/boards/1?selectedIssue=KAN-13)

## Motivation and Context

These changes help to establish the frontend workflow and environment by allowing any developer to check their code 
against the ESLint / Prettier standard on their own machine.

## How Has This Been Tested?

These changes have been tested locally on my system using VS Code.
Steps to Reproduce (MacOS):
- `brew install node`
- `npm install eslint@8.57.0  -g --save-dev`
- `npm install eslint-config-prettier --save-dev`
- `npm install jest --save-dev`

**Testing:**
- Run application on localhost:3001 `npm start`
- Run Linter: `npm run lint`
- Run Linter with autocorrection: `npm run lint -- --fix`

**Docker:**

Successfully built a local Docker Image at christianmckinnon/jhaiapp1.0

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests to cover my changes.
- [x] I have passed lints.
- [x] All new and existing tests passed.

## Further comments

None.

### CODE REVIEW

File: .github/pull_request_template.md
Comment (MasakiNishi) [inline .github/pull_request_template.md:1]: Can this file be restored to its original template state?

This template is automatically recognized when a pull request is created and pasted into the pull request summary field. If you change it, the modified state will be pasted as the template.

File: README.md [hunk 1]
```diff
@@ -1,46 +1,39 @@
...
-
-You don’t have to ever use `eject`. The curated feature set is suitable for small and middle deployments, and you shouldn’t feel obligated to use this feature. However we understand that this tool wouldn’t be useful if you couldn’t customize it when you are ready for it.
-
-## Learn More
-
-You can learn more in the [Create React App documentation](https://facebook.github.io/create-react-app/docs/getting-started).
+### `npm start`
 
-To learn React, check out the [React documentation](https://reactjs.org/).
+After running, please open [http://localhost:8080](http://localhost:8080) to view the application in the browser.
```
Comment (MasakiNishi) [inline README.md]: It is running on the 3001 port now. So, can you change this or can you remove it since it is written in infra repo?

**Review by MasakiNishi — CHANGES_REQUESTED** [review]
Thanks for setting up eslint, and I can confirm that it works with docker-compose.

It would be very helpful if you could change only the small modifications I commented on!

File: .github/pull_request_template.md
Comment (christianmckinnon) [reply .github/pull_request_template.md:1]: That makes sense. I have made the change.

File: README.md (hunk 1 above)
Comment (christianmckinnon) [reply README.md]: Yes, I forgot to update this! It's been removed.

**Review by MasakiNishi — APPROVED** [review]
Thank you so much! I will merge this! I really appreciate this!

**Pull request merged**